conslogging/conslogging.go

```diff
@@ -467,11 +468,29 @@ func (cl ConsoleLogger) color(c *color.Color) *color.Color {
 	return noColor
 }
 
+var (
+	// githubRegExp Matches :2dd88e53f2e59e96ec1f9215f24a3981e5565edf+ in a prefix.
+	// 	Prefix containing hash may resemble: g/e/hello-world:2dd88e53f2e59e96ec1f9215f24a3981e5565edf+base
+	//	Prefix must be exactly 40 characters
+	githubRegExp = regexp.MustCompile(`:[a-z0-9]{40}\+`)
+
+	branchOrSHARegExp = regexp.MustCompile(`:[a-zA-Z0-9/\-_]*\+`)
+)
+
 func prettyPrefix(prefixPadding int, prefix string) string {
-	if prefixPadding == NoPadding {
+	if prefixPadding <= NoPadding {
 		return prefix
 	}
 
+	// tempReplacement used to temporarily remove branch name / sha from string
+	const tempReplacement = "{{}}{{{}{}}}"
```

Comment:
this feels fragile, I think it would be better to check if `githubRegExp` matches the input, and then if so do a replacement.

if that regex doesn't match, then no replacements should be required.

Reply:
This particular replacement change is meant to tackle when the branch name has special characters; specifically, a `/`. I just use it in both scenarios.

The previous version of this function would cut off a branch name like `ryan/implement-feature` due to the `/`.
I'll admit I took a shortcut here - doing this rather than refactoring the entire function.

It looks odd - hacky even - though I'm not sure it's fragile. Could you elaborate?

Reply:
Refactored to remove the tempReplacement hack